eZ/Publish/API/Repository/Tests/FieldType/UserIntegrationTest.php

```diff
@@ -466,21 +493,37 @@ public function testRemoveFieldDefinition()
 
     public function testAddFieldDefinition()
     {
-        $this->markTestIncomplete(
-            'Currently cannot be tested since user can be properly created only through UserService'
-        );
+        // Field cannot be added to ContentType with existing content.
+        $this->expectException(BadStateException::class);
+
+        return parent::testAddFieldDefinition();
     }
 
     /**
-     * @param mixed $failingValue
-     * @param string $expectedException
-     *
-     * @dataProvider provideInvalidUpdateFieldData
+     * @depends testCreateContent
      */
-    public function testUpdateContentFails($failingValue, $expectedException)
+    public function testCopyField($content)
     {
-        $this->markTestIncomplete(
-            'Currently cannot be tested since user can be properly created only through UserService'
-        );
+        // Users cannot be copied.
+        $this->expectException(ForbiddenException::class);
+        $this->expectExceptionMessage('User "hans" already exists');
+
+        return parent::testCopyField($content);
+    }
+
+    /**
+     * @depends testCopyField
+     */
+    public function testCopiedFieldType($content)
+    {
+        $this->markTestSkipped('Users cannot be copied, content is not passed to test.');
```

Comment:
> I have no idea how to improve this but I am against marking a test as skipped by design. We have already too much-skipped tests which can be dangerous and extend the risk of error

Those tests have a common base for a Field Type, however User Field type strategy doesn't allow this because of behavior shown in `testCopyField`. I tried to implement here something more useful, including dropping those methods overrides, but it's technically not possible.

Skipping tests is not bad by design. We skip too much, but here is a good example of correct, strategy-based, implementation. At least from my POV.